pwnlib/shellcraft/templates/amd64/windows/winexec.asm

```diff
@@ -7,15 +7,19 @@
 
 Args:
     cmd (str): The program to execute.
+    cmd_show (int): nCmdShow parameter.
 </%docstring>
-<%page args="cmd"/>
+<%page args="cmd, cmd_show = 0"/>
 <%
 cmd = _need_bytes(cmd)
 %>
 
     ${amd64.windows.getprocaddress(b'WinExec', b'kernel32.dll', 'rsi')}
     ${amd64.pushstr(cmd)}
     mov rcx, rsp
+    and rsp, -16
     sub rsp, 0x30
+    ${amd64.mov('rdx', cmd_show)}
     call rsi
-    add rsp, ${pretty(0x30+align(8, len(cmd)))}
+    add rsp, ${pretty(0x30+align(8, len(cmd)+1))}
+    or rsp, 8
```

Comment:
This seems inverted now and `rsp` doesn't land on the old value before the `pushstr` neither on `cmd.exe` nor `calc.exe`.

I'm testing this under linux by commenting out the `getprocaddress` and `call rsi` instructions and running `shellcraft -d amd64.windows.winexec calc.exe` in a shell if that makes a difference.